### PR TITLE
method get_sha_from_tag

### LINKED ISSUE #170: Implement GitLab method: get_sha_from_tag
```python
    def get_sha_from_tag(self, tag_name: str) -> str:
        raise NotImplementedError()
```

- https://python-gitlab.readthedocs.io/en/stable/gl_objects/commits.html

### PR DESCRIPTION
**Pull request by lbarcziova**
Fixes #170

### CODE REVIEW

File: ogr/services/gitlab.py [hunk 1]
```diff
@@ -221,7 +221,12 @@ def get_pr_list(self, status: PRStatus = PRStatus.open) -> List["PullRequest"]:
         return [self._pr_from_gitlab_object(mr) for mr in mrs]
 
     def get_sha_from_tag(self, tag_name: str) -> str:
-        raise NotImplementedError()
+        try:
+            tag = self.gitlab_repo.tags.get(tag_name)
+            return tag.attributes["commit"]["id"]
+        except gitlab.exceptions.GitlabGetError:
+            logger.error(f"Tag {tag_name} was not found.")
+            return ""
```
Comment (lachmanfrantisek) [inline ogr/services/gitlab.py]: Maybe it would be better to raise an exception?

Comment (lachmanfrantisek) [reply ogr/services/gitlab.py]:
```suggestion
            raise PackitException(f"Tag {tag_name} was not found.", ex)
```

File: tests/integration/test_gitlab.py [hunk 2]
```diff
@@ -66,6 +66,13 @@ def test_email(self):
         assert "@" in email
         assert "." in email
 
+    def test_get_sha_from_tag(self):
+        assert (
+            self.project.get_sha_from_tag("0.1.0")
+            == "957d267a5b0cd9e615cd081c0eb02397dce1eb73"
+        )
+        assert not self.project.get_sha_from_tag("future")
```
Comment (lachmanfrantisek) [inline tests/integration/test_gitlab.py]:
```suggestion
        with pytest.raises(PackitException) as ex:
            self.project.get_sha_from_tag("future")
        assert "not found" in str(ex.value)
```

File: ogr/services/gitlab.py (hunk 1 above)
Comment (lachmanfrantisek) [reply ogr/services/gitlab.py]: Could you fix that also for github please?

Comment (lbarcziova) [reply ogr/services/gitlab.py]: Why **PackitException**?

Comment (lachmanfrantisek) [reply ogr/services/gitlab.py]: Ups. `OgrException` (or ideally newly-created `GitlabAPIException`) would be of course better.

File: tests/integration/test_github.py [hunk 3]
```diff
@@ -149,7 +151,8 @@ def test_get_sha_from_tag(self):
             self.ogr_project.get_sha_from_tag("0.0.1")
             == "29ca3caefc781b4b41245df3e01086ffa4b4639e"
         )
-        assert not self.ogr_project.get_sha_from_tag("future")
+        with pytest.raises(GithubAPIException):
```
Comment (lachmanfrantisek) [inline tests/integration/test_github.py]:
```suggestion
        with pytest.raises(GithubAPIException) as ex:
```

File: tests/integration/test_github.py [hunk 4]
```diff
@@ -149,7 +151,8 @@ def test_get_sha_from_tag(self):
...
+            self.ogr_project.get_sha_from_tag("future")
```
Comment (lachmanfrantisek) [inline tests/integration/test_github.py]:
```suggestion
            self.ogr_project.get_sha_from_tag("future")
        assert "not found" in str(ex.value)
```

**Comment by lachmanfrantisek** [comment]
recheck

**Comment by lachmanfrantisek** [comment]
recheck

**Comment by lachmanfrantisek** [comment]
recheck

**Comment by lachmanfrantisek** [comment]
recheck

**Pull request merged**